cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_node.go

```diff
@@ -17,20 +17,36 @@ limitations under the License.
 package bootstrap
 
 import (
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
 	iamv1 "sigs.k8s.io/cluster-api-provider-aws/cmd/clusterawsadm/api/iam/v1alpha1"
 )
 
-func (t Template) secretPolicy() iamv1.StatementEntry {
-	return iamv1.StatementEntry{
-		Effect: iamv1.EffectAllow,
-		Resource: iamv1.Resources{
-			"arn:*:secretsmanager:*:*:secret:aws.cluster.x-k8s.io/*",
-		},
-		Action: iamv1.Actions{
-			"secretsmanager:DeleteSecret",
-			"secretsmanager:GetSecretValue",
-		},
+func (t Template) secretPolicy(secureSecretsBackend infrav1.SecretBackend) iamv1.StatementEntry {
+	switch secureSecretsBackend {
```

Comment:
This is very similar to  cluster_api_controller.go above. Do you think it would be possible to consolidate into 1 place and add the additional actions for the controller policy?

Not essential though.

Reply:
This kinda thing pops up a lot in the clusterctl code, I think it can be handled in a separate PR as a mini refactor?